### PR TITLE
fix: ESLint for base JS files

### PR DESCRIPTION
**Pull request by Lucky38i**
## Summary of Changes

Currently VSCode reads from eslintignore whereas IntelliJ reads from eslintrc.js. This presents issues for users working inside of the base JS files of A32NX where they get no linting at all when using VSCode. 

This PR aims to ensure regardless of IDE there correct linting rules are adhered to.

There is also a custom eslintrc for base JS files that aims to maintain a useful level of rules for these files without require a rewrite of most classes.

## Additional context

Discord username (if different from GitHub): Lucky38i
## Testing instructions

Doesn't require Testing

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -7,7 +7,32 @@
...
+    "fbw-a32nx/src/systems/fmgc/src/flightplanning/WaypointBuilder.ts",
+    "fbw-a32nx/src/systems/fmgc/src/flightplanning/WorldMagneticModel.ts",
+    "fbw-a32nx/src/systems/fmgc/src/utils/LzUtf8.js",
+    "target/*",
+    "CMakeFiles/*",
+    "cmake*",
+    "fbw-a32nx/out/*",
+    "fbw-a380x/out/*"
   ],
   // DO NOT CHANGE THIS FILE (unless you are an admin) (if admin, make sure you change the same file in the latest asobo branch as well)
```
Comment (frankkopp) [inline .eslintrc:37]: This can be removed or changed I guess :)

**Review by frankkopp — APPROVED** [review]
If this still works for the VSCode users it lgtm

File: .eslintrc (hunk 1 above)
Comment (Lucky38i) [reply .eslintrc:37]: I don't think we want linting on our generated files

Comment (frankkopp) [reply .eslintrc:37]: I meant the comment `// DO NOT....`

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = {
+    rules: {
+        'strict': 'off',
+        'no-unused-vars': 'warn',
+        '@typescript-eslint/no-unused-vars': 'warn',
+        'no-underscore-dangle': 'warn',
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/.eslintrc.js:8]: I suggest removing this. A ton of the Asobo files use leading underscores heavily. This rule also makes getters and setters a bit awkward in general.

**Review by tracernz — COMMENTED** [review]
How do these rules look on the existing code? We might need to tweak them a bit to suit the existing conventions.

It's always worked fine in vscode for me without these changes, at least for brace style etc. Not sure what's doing it 😆 .

**Review by tracernz — DISMISSED** [review]
Looking at the fixup commit, the rules are wrong. Our convention is double quotes for strings in JS files.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "../../../../.eslintrc",
```
Comment (Benjozork) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/.eslintrc:2]: Why not just put the rules in the root `.eslintrc` here? They are only meant for `base/` anyway.

Comment (tracernz) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/.eslintrc:2]: The A380 should not use any legacy JS.

**Pull request closed without merging**